queries.sql

```diff
@@ -23,3 +23,221 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
+FROM animals a
+JOIN owners o ON a.owner_id = o.id
+WHERE o.full_name = 'Melody Pond';
+
+-- List of all animals that are Pokémon (their type is Pokémon).
+SELECT a.name
+FROM animals a
+JOIN species s ON a.species_id = s.id
+WHERE s.name = 'Pokemon';
+
+-- List all owners and their animals, including those who don't own any animals.
+SELECT o.full_name, a.name
+FROM owners o
+LEFT JOIN animals a ON o.id = a.owner_id;
+
+-- How many animals are there per species?
+SELECT s.name AS species, COUNT(a.id) AS animal_count
+FROM species s
+LEFT JOIN animals a ON s.id = a.species_id
+GROUP BY s.name;
+
+-- List all Digimon owned by Jennifer Orwell.
+SELECT a.name
+FROM animals a
+JOIN owners o ON a.owner_id = o.id
+JOIN species s ON a.species_id = s.id
+WHERE o.full_name = 'Jennifer Orwell' AND s.name = 'Digimon';
+
+-- List all animals owned by Dean Winchester that haven't tried to escape.
+SELECT a.name
+FROM animals a
+JOIN owners o ON a.owner_id = o.id
+WHERE o.full_name = 'Dean Winchester' AND a.escape_attempts = 0;
+
+-- Who owns the most animals?
+SELECT o.full_name, COUNT(a.id) AS animal_count
+FROM owners o
+LEFT JOIN animals a ON o.id = a.owner_id
+GROUP BY o.full_name
+ORDER BY animal_count DESC
+LIMIT 1;
+
+
+-- 1. Who was the last animal seen by William Tatcher?
+SELECT a.name
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets vet ON v.vet_id = vet.id
+WHERE vet.name = 'Vet William Tatcher'
+ORDER BY v.visit_date DESC
+LIMIT 1;
+
+-- 2. How many different animals did Stephanie Mendez see?
+SELECT COUNT(DISTINCT a.id)
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets vet ON v.vet_id = vet.id
+WHERE vet.name = 'Vet Stephanie Mendez';
+
+-- 3. List all vets and their specialties, including vets with no specialties.
+SELECT v.name AS vet_name, COALESCE(s.name, 'No Specialty') AS specialty
+FROM vets v
+LEFT JOIN specializations sp ON v.id = sp.vet_id
+LEFT JOIN species s ON sp.species_id = s.id;
+
+-- 4. List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT a.name
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets vet ON v.vet_id = vet.id
+WHERE vet.name = 'Vet Stephanie Mendez'
+AND v.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
+
+-- 5. What animal has the most visits to vets?
+SELECT a.name AS animal_name, COUNT(*) AS visit_count
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+GROUP BY a.name
+ORDER BY visit_count DESC
+LIMIT 1;
+
+-- 6. Who was Maisy Smith's first visit?
+SELECT a.name AS animal_name
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets vet ON v.vet_id = vet.id
+WHERE vet.name = 'Vet Maisy Smith'
+ORDER BY v.visit_date
+LIMIT 1;
+
+-- 7. Details for the most recent visit: animal information, vet information, and date of visit.
+SELECT a.name AS animal_name, vet.name AS vet_name, v.visit_date
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets vet ON v.id = v.vet_id
+ORDER BY v.visit_date DESC
+LIMIT 1;
```

Comment:
- [ ] It would be better if we could display properly the results of this part of the query. Currently, this query results have not been shown properly, let us kindly check it and update it accordingly
![Screenshot from 2023-10-25 08-22-38](https://github.com/Rifat8080/vet-clinic-backend-app/assets/46388113/6fc46b65-1ba4-457d-a69d-c8a860f50f21)
![Screenshot from 2023-10-25 08-29-37](https://github.com/Rifat8080/vet-clinic-backend-app/assets/46388113/03093790-41a0-479a-8a57-18a5adb605d3)

Reply:
<img width="1032" alt="Screenshot 2023-10-25 at 3 40 59 PM" src="https://github.com/Rifat8080/vet-clinic-backend-app/assets/123451855/9bde4732-faff-471c-9d68-ab3349b7e89c">